src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -336,14 +336,15 @@ private function isMaxDepthReached($class, $attribute, array &$context)
     /**
      * Gets the cache key to use.
      *
-     * @param array $context
+     * @param string|null $format
+     * @param array       $context
      *
      * @return bool|string
      */
-    private function getCacheKey(array $context)
+    private function getCacheKey($format, array $context)
     {
         try {
-            return md5(serialize($context));
+            return $format.'-'.md5(serialize($context));
```

Comment:
It's already a string and it makes the key more explicit. On the other hand if the format name is very long, it can hit the max key size. What do you think?